src/com/amazon/ionelement/impl/StructElementImpl.kt

```diff
@@ -121,43 +122,40 @@ internal class StructElementImpl(
 
     override fun equals(other: Any?): Boolean {
         if (this === other) return true
-        if (other !is StructElementImpl) return false
+        if (other !is StructElement) return false
```

Comment:
🗺️ This change had quite a large effect. The prior implementation use a private field, so I had to rewrite it to not use that field.